analytics_data_api/v0/tests/views/test_learners.py

```diff
@@ -404,9 +412,192 @@ def test_pagination(self):
+            'engagement_ranges': {
+                'date_range': {
+                    'start': None,
+                    'end': None
+                }
+            }
+        }
+        empty_range = {
+            range_type: [None, None] for range_type in ['below_average', 'average', 'above_average']
+        }
+        for metric in self.engagement_metrics:
+            empty_engagement_ranges['engagement_ranges'][metric] = copy.deepcopy(empty_range)
+        return empty_engagement_ranges
+
+    @property
+    def engagement_metrics(self):
+        """ Convenience method for getting the metric types. """
+        metrics = []
+        for entity_type in engagement_entity_types.AGGREGATE_TYPES:
+            for event in engagement_events.EVENTS[entity_type]:
+                metrics.append('{0}_{1}'.format(entity_type, event))
+        return metrics
+
+    # TODO: enable during https://openedx.atlassian.net/browse/AN-6319
+    # def test_cohorts(self):
+    #     pass
+
+    def test_no_engagement_ranges(self):
+        response = self._get(self.course_id)
+        self.assertEqual(response.status_code, 200)
+        self.assertDictContainsSubset(self.empty_engagement_ranges, json.loads(response.content))
+
+    def test_one_engagement_range(self):
+        metric_type = '{0}_{1}'.format(engagement_entity_types.PROBLEMS, engagement_events.COMPLETED)
+        start_date = datetime.datetime(2015, 7, 1, tzinfo=pytz.utc)
+        end_date = datetime.datetime(2015, 7, 21, tzinfo=pytz.utc)
+        G(ModuleEngagementMetricRanges, course_id=self.course_id, start_date=start_date, end_date=end_date,
+          metric=metric_type, range_type='high', low_value=90, high_value=6120)
+        expected_ranges = self.empty_engagement_ranges
+        expected_ranges['engagement_ranges'].update({
+            'date_range': {
+                'start': '2015-07-01',
+                'end': '2015-07-21'
+            },
+            metric_type: {
+                'below_average': [None, None],
+                'average': [None, 90.0],
+                'above_average': [90.0, 6120.0]
+            }
+        })
+
+        response = self._get(self.course_id)
+        self.assertEqual(response.status_code, 200)
+        self.assertDictContainsSubset(expected_ranges, json.loads(response.content))
+
+    def _get_full_engagement_ranges(self):
+        """ Populates a full set of engagement ranges and returns the expected engagement ranges. """
+        start_date = datetime.datetime(2015, 7, 1, tzinfo=pytz.utc)
+        end_date = datetime.datetime(2015, 7, 21, tzinfo=pytz.utc)
+
+        expected = {
+            'engagement_ranges': {
+                'date_range': {
+                    'start': '2015-07-01',
+                    'end': '2015-07-21'
+                }
+            }
+        }
+
+        max_value = 1000.0
+        for metric_type in self.engagement_metrics:
+            low_ceil = random.random() * max_value * 0.5
```

Comment:
I'd prefer if we didn't use the `random` module for values in tests, because I don't like the idea of nondeterministic tests. If you're concerned about a variety of values, then you can parameterize this method and use `ddt` in the test cases for particularly problematic values, but if there's no reason for variance in these values, then I don't see a problem with hardcoding them in.